File: string/127.cpp
#include <bits/stdc++.h>

using namespace std;

bool check_duplicate_parentheses(const string &cad) {
    stack<char> st;
    int cont = 0;

    for (const char &ch : cad) {
        if (ch == ')' && st.top() == ')') {
            return true;
        } else {
            st.push(ch);
        }
    }

    return false;
}

int main() {
    string arr[] = {"((x+y))+z", "(x+y)", "((x+y)+((z)))"};
    for (int i = 0; i < sizeof(arr) / sizeof(arr[0]); ++i) {
        cout << boolalpha << check_duplicate_parentheses(arr[i]) << "\n";
    }

    return 0;
}

File: string/127.py
def check_duplicate_parentheses(cad: str):
    stack = []

    for ch in cad:
        if len(stack) > 0 and ch == ")" and stack[-1] == ")":
            return True
        else:
            stack.append(ch)

    return False


if __name__ == "__main__":
    cads = ["((x+y))+z", "(x+y)", "((x+y)+((z)))"]

    for cad in cads:
        print(check_duplicate_parentheses(cad))
